### PR TITLE
php 8.0.0

### PR DESCRIPTION
**Pull request by derrabus**
~~**Do not merge yet**: Official announcement pending~~

-----

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This PR suggests to bump php to version 8.0.0 which ~~should be~~ has been released today. ~~We need to wait until the official announcement, but I wanted to submit the formula already, so we can run the build through the CI and discuss the changes I've applied to the formula itself.~~

~~If the SHA256 sum changes before the official announcement, I'll update the PR. If #65689 gets merged first (which is fine), I'll rebase.~~ Done.

### CODE REVIEW

**Comment by derrabus** [comment]
PHP 8.0.0 is official now. The SHA256 on https://www.php.net/downloads.php is the same as in this PR.

PR is ready. 🎉

**Comment by sukei** [comment]
Feuuuuu !

File: Formula/php.rb [hunk 1]
```diff
@@ -64,9 +64,6 @@ class Php < Formula
   patch :DATA
 
   def install
-    # Ensure that libxml2 will be detected correctly in older MacOS
-    ENV["SDKROOT"] = MacOS.sdk_path if MacOS.version == :el_capitan || MacOS.version == :sierra
-
```
Comment (SMillerDev) [inline Formula/php.rb:69]: Is this fixed now?

**Review by SMillerDev — COMMENTED** [review]
Can you try and submit these patches upstream (again)?

File: Formula/php.rb (hunk 1 above)
Comment (derrabus) [reply Formula/php.rb:69]: I don't have a El Capitan or Sierra environment to test this. My understanding is that those versions are not supported by Homebrew anymore. But I can re-add the line. It's an unrelated change, after all.

Comment (nabihnadhif) [reply Formula/php.rb:69]: @derrabus El Capitan and Sierra are supported on a best-effort basis. see https://docs.brew.sh/Installation#2

> 10.14 or higher is recommended. 10.9–10.13 are supported on a best-effort basis.

Comment (derrabus) [reply Formula/php.rb]: Thanks @nabihnadhif. I have reverted this change.

**Comment by derrabus** [comment]
@SMillerDev I have removed the `patch :DATA` call locally and the whole patch block at the end of the formula and the formula does compile and passes `brew test` on my machine (Intel, Big Sur, CLT 12.0). 🤔

**Comment by derrabus** [comment]
Strange. With the patch removed, the Mojave and Catalina builds pass, but the Big Sur build fails. I've reverted to the previous state. Let's do the bump first and figure out how much of the patch we still need afterwards.

**Review by nabihnadhif — APPROVED** [review]

**Review by SMillerDev — APPROVED** [review]
Thanks @derrabus  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by derrabus** [comment]
@SMillerDev The PR still has a `do not merge` label. I suppose, @BrewTestBot won't pick it up for merge in that case, right? Can you remove it?

**Comment by SMillerDev** [comment]
A human needs to merge this, and this human will do it as soon as he's done with the meetings he's in.

**Comment by ruudk** [comment]
Thanks for working on this! 

Now that this is merged, it's no longer possible to install `php@7.4`

**Comment by antiseptikk** [comment]
Thanks for adding php 8.0, but it's very annoying for php@7.4 because the formula doesn't work anymore.

**Comment by ruudk** [comment]
I'm adding it back here https://github.com/Homebrew/homebrew-core/pull/65779

**Pull request closed without merging**